packages/extension/src/newtab/DndModal.tsx

```diff
@@ -1,95 +1,104 @@
-import { Dropdown } from '../../../shared/src/components/fields/Dropdown';
-import { TextField } from '../../../shared/src/components/fields/TextField';
-import { ModalCloseButton } from '../../../shared/src/components/modals/ModalCloseButton';
+import React, { FC, useState, useContext } from 'react';
+import { format } from 'date-fns';
+import { Radio } from '@dailydotdev/shared/src/components/fields/Radio';
+import { Dropdown } from '@dailydotdev/shared/src/components/fields/Dropdown';
+import { TextField } from '@dailydotdev/shared/src/components/fields/TextField';
+import { ModalCloseButton } from '@dailydotdev/shared/src/components/modals/ModalCloseButton';
 import {
   StyledModal,
   ModalProps,
-} from '../../../shared/src/components/modals/StyledModal';
-import styles from './DoNotDisturbModal.module.css';
+} from '@dailydotdev/shared/src/components/modals/StyledModal';
+import { getDefaultLink, dndOption, CustomTime, TimeFormat } from './dnd';
 import DndContext from './DndContext';
 
-export interface DoNotDisturbModalProps extends ModalProps {
-  defaultTimeFormat?: TimeFormat;
-}
-
-const timeFormatOptions = Object.values(TimeFormat).map((value) => ({
-  label: value,
-  value,
+const timeFormatOptions = Object.entries(dndOption).map(([k, v]) => ({
+  label: v.label,
+  value: k,
 }));
-
 const customTimeOptions = Object.values(CustomTime);
 
-const DoNotDisturbModal: React.FC<DoNotDisturbModalProps> = ({
-  defaultTimeFormat = TimeFormat.HALF_HOUR,
+const DoNotDisturbModal: FC<ModalProps> = ({
   onRequestClose,
   ...modalProps
 }) => {
-  const { setDndSettings, isActive } = React.useContext(DndContext);
-  const [link, setLink] = React.useState('');
-  const [customNumber, setCustomNumber] = React.useState(0);
-  const [customTimeIndex, setCustomTimeIndex] = React.useState(0);
-  const [dndTime, setDndTime] = React.useState(defaultTimeFormat);
+  const { dndSettings, setDndSettings, isActive } = useContext(DndContext);
+  const [link, setLink] = useState('');
+  const [customNumber, setCustomNumber] = useState(0);
+  const [customTimeIndex, setCustomTimeIndex] = useState(0);
+  const [dndTime, setDndTime] = useState<TimeFormat>('HALF_HOUR');
 
   const handleSubmit = async (e: React.MouseEvent<Element, MouseEvent>) => {
-    const expiration =
-      dndTime === TimeFormat.CUSTOM
-        ? getExpiration(customTimeOptions[customTimeIndex], customNumber)
-        : getTimeFormatExpiration(dndTime);
-
+    const option = dndOption[dndTime];
+    const customTime = customTimeOptions[customTimeIndex];
+    const expiration = option.getExpiration(customTime, customNumber);
     const fallback = link || getDefaultLink();
 
     await setDndSettings({ expiration, link: fallback });
 
     onRequestClose(e);
   };
 
-  const renderForm = () => {
-    if (isActive) return <br />;
+  const turnedOnContent = (
+    <div className="grid grid-cols-2 gap-6 justify-center my-6">
+      <button
```

Comment:
We have a custom tailwind configuration that increases the reusability of classes and a custom `Button` component.
For example, the Cancel button class is `btn-secondary` with a `buttonSize` property set to `large`.
It automatically sets the border-radius, border color, size, typography, and everything else basically.
Please take a look at other components to see how we use the `Button` component.